### PR TITLE
Refactor assets and common styles

### LINKED ISSUE #28: Refactor out common styles between screens
We have a lot of duplicate stylings between different screens. Look into creating general components.

### PR DESCRIPTION
**Pull request by einaru**
Closes #28

### CODE REVIEW

File: core/src/main/kotlin/no/ntnu/beardblaster/screen/BeardBlasterStage.kt [hunk 1]
```diff
@@ -3,15 +3,15 @@ package no.ntnu.beardblaster.screen
 import com.badlogic.gdx.Gdx
 import com.badlogic.gdx.scenes.scene2d.Stage
 import com.badlogic.gdx.utils.viewport.FitViewport
-import no.ntnu.beardblaster.worldHeight
-import no.ntnu.beardblaster.worldWidth
+import no.ntnu.beardblaster.HEIGHT
+import no.ntnu.beardblaster.WIDTH
 
-class BeardBlasterStage : Stage(FitViewport(worldWidth, worldHeight)) {
+class BeardBlasterStage : Stage(FitViewport(WIDTH, HEIGHT)) {
```
Comment (cherrybeans) [inline core/src/main/kotlin/no/ntnu/beardblaster/screen/BeardBlasterStage.kt]: Not a big thing, but perhaps use "WORLD_HEIGHT" and "WORLD_WIDTH" to make it more explicit?

File: core/src/main/kotlin/no/ntnu/beardblaster/screen/GameplayScreen.kt [hunk 2]
```diff
@@ -1,105 +1,76 @@
...
+private val log = logger<GameplayScreen>()
+
+class GameplayScreen(
+    game: BeardBlasterGame,
+    batch: Batch,
+    assets: AssetStorage,
+    camera: OrthographicCamera,
+) : BaseScreen(game, batch, assets, camera) {
+    private val attackBtn = scene2d.textButton(Nls.attack())
+    private val quiteBtn = scene2d.textButton(Nls.quit())
```
Comment (cherrybeans) [inline core/src/main/kotlin/no/ntnu/beardblaster/screen/GameplayScreen.kt:28]: Love it!!

File: core/src/main/kotlin/no/ntnu/beardblaster/screen/GameplayScreen.kt [hunk 3]
```diff
@@ -1,105 +1,76 @@
...
+        attackBtn.onClick {
+            log.info { Nls.wizardNAttacks(1) }
         }
-        btnQuit.onClick {
+        quiteBtn.onClick {
             game.removeScreen<GameplayScreen>()
-            game.addScreen(GameplayScreen(game))
+            // FIXME Really needed to add a new game play screen here?
+            // Might be better to add it when joining/starting a new game.
+            game.addScreen(GameplayScreen(game, batch, assets, camera))
```
Comment (cherrybeans) [inline core/src/main/kotlin/no/ntnu/beardblaster/screen/GameplayScreen.kt:59]: Agree with this thought. We can move adding the gameplay screen to LobbyScreen

File: core/src/main/kotlin/no/ntnu/beardblaster/screen/LoadingScreen.kt [hunk 4]
```diff
@@ -1,78 +1,84 @@
...
-            it.color = Color.RED
-            it.rect(50f, (cam.viewportHeight / 2) - 12f, progress * (cam.viewportWidth - 50), 25f)
+        val x = 200f
+        val height = 25f
+        val width = camera.viewportWidth - (x * 2)
+        val y = (camera.viewportHeight / 2) - (height / 2)
+        val progress = assets.progress.percent
+
+        renderer.use(ShapeRenderer.ShapeType.Filled, camera.combined) {
+            it.color = Color.PINK
```
Comment (cherrybeans) [inline core/src/main/kotlin/no/ntnu/beardblaster/screen/LoadingScreen.kt:46]: The most important thing thing I've seen in this PR

File: core/src/main/kotlin/no/ntnu/beardblaster/screen/LoadingScreen.kt [hunk 5]
```diff
@@ -1,78 +1,84 @@
...
+    private fun addGameScreens() {
+        game.addScreen(LoginMenuScreen(game, batch, assets, camera))
+        game.addScreen(LoginScreen(game, batch, assets, camera))
+        game.addScreen(RegisterScreen(game, batch, assets, camera))
+        game.addScreen(MenuScreen(game, batch, assets, camera))
+        game.addScreen(TutorialScreen(game, batch, assets, camera))
+        game.addScreen(HighScoreScreen(game, batch, assets, camera))
+        game.addScreen(LobbyScreen(game, batch, assets, camera))
+        game.addScreen(JoinLobbyScreen(game, batch, assets, camera))
+        game.addScreen(GameplayScreen(game, batch, assets, camera))
```
Comment (cherrybeans) [inline core/src/main/kotlin/no/ntnu/beardblaster/screen/LoadingScreen.kt:76]: Can remove this from here and only add it in LobbyScreen when starting game

File: core/src/main/kotlin/no/ntnu/beardblaster/screen/BeardBlasterStage.kt (hunk 1 above)
Comment (einaru) [reply core/src/main/kotlin/no/ntnu/beardblaster/screen/BeardBlasterStage.kt]: Agree 👍🏻

File: core/src/main/kotlin/no/ntnu/beardblaster/screen/LoadingScreen.kt (hunk 5 above)
Comment (einaru) [reply core/src/main/kotlin/no/ntnu/beardblaster/screen/LoadingScreen.kt:76]: That make sense

Comment (einaru) [reply core/src/main/kotlin/no/ntnu/beardblaster/screen/LoadingScreen.kt:76]: On the other hand we need, for example,  access to the `HighScoreScreen` from the `MenuScreen`. I think I'll leave this as it is, and then we can refactor at a later stage.

File: core/src/main/kotlin/no/ntnu/beardblaster/screen/GameplayScreen.kt (hunk 3 above)
Comment (einaru) [reply core/src/main/kotlin/no/ntnu/beardblaster/screen/GameplayScreen.kt:59]: Tried to move `addScreen` within `startButton.OnClick` in the `LobbyScreen`, but that resulted in:

```
com.badlogic.gdx.utils.GdxRuntimeException: Screen already registered to type: class no.ntnu.beardblaster.screen.GameplayScreen.
```

Leaving it as it is, until we figure out how to handle it.

**Review by gullskatten — APPROVED** [review]

**Review by eirikgm — APPROVED** [review]

**Pull request merged**